src/screen.c

```diff
@@ -4914,6 +4917,43 @@ set_chars_option(win_T *wp, char_u *value, int is_listchars, int apply,
 		    break;
 		}
 
+		if (!is_listchars && STRCMP(tab[i].name.string, "ellipsis") == 0)
+		{
+		    if (round == 0)
+		    {
+			ellipsis_len = 0;
+			while (*s != NUL && *s != ',')
+			{
+			    c1 = get_encoded_char_adv(&s);
+			    ++ellipsis_len;
+			}
+			if (ellipsis_len == 0)
+			    return field_value_err(errbuf, errbuflen,
+				e_wrong_number_of_characters_for_field_str,
+				tab[i].name.string);
+		    }
+		    else
+		    {
+			vim_free(fill_chars.ellipsis);
+			fill_chars.ellipsis = ALLOC_MULT(int, ellipsis_len + 1);
+
+			if (fill_chars.ellipsis != NULL)
+			{
+			    int pos = 0;
+			    while (*s != NUL && *s != ',')
+			    {
+				c1 = get_encoded_char_adv(&s);
+				fill_chars.ellipsis[pos++] = c1;
+			    }
+			    fill_chars.ellipsis[pos] = NUL;
+			}
+		    }
+
+		    p = s;
```

Comment:
Also move the `p = s` out of the `if` and `else` branches for `multispace` and `leadmultispace`?